core/src/main/java/io/questdb/cutlass/http/HttpConnectionContext.java

```diff
@@ -420,12 +424,17 @@ private boolean configureSecurityContext() {
             if (!authenticator.authenticate(headerParser)) {
                 return false;
             }
-            securityContext = configuration.getFactoryProvider().getSecurityContextFactory().getInstance(
-                    authenticator.getPrincipal(),
-                    authenticator.getAuthType(),
-                    SecurityContextFactory.HTTP
-            );
-            securityContext.authorizeHttp();
+            try {
+                securityContext = securityContextFactory.getInstance(
+                        authenticator.getPrincipal(),
+                        authenticator.getAuthType(),
+                        SecurityContextFactory.HTTP
+                );
+                securityContext.authorizeHttp();
+            } catch (CairoException e) {
```

Comment:
Why the todo(s) ? Isn't CairoException thrown when access is denied ?

Reply:
as it is now, the error message will say something like 'rejecting unauthenticated request...`.
however, authentication actually succeeds.
the problem is that the user does not have HTTP permission.

letting the CairoException propagated upstream just simply disconnects the client, instead of sending back the error message.
so will need to handle the exception nicely somewhere, and create a proper HTTP response with the error message taken from the exception.
the todo is about sending back a proper error message to the client.
this change quickly addresses the disconnect problem, but will need further work on another PR.